.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,159 @@
+name: Nightly build
+
+on:
+  push:
+    branches: master
+  schedule:
+    - cron: "0 1 * * *"
+
+jobs:
+
+  dockerfile-lint:
+
+    if: github.repository_owner == 'Autoware-AI'
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+      fail-fast: false
+
+    container:
+      image: hadolint/hadolint:latest-debian
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Linting
+        run: |
+          hadolint generic/Dockerfile
+          hadolint generic/Dockerfile.legacy.catkin
+          hadolint generic/Dockerfile.legacy.colcon
+          hadolint generic/Dockerfile.base
+          hadolint generic/Dockerfile.cuda.kinetic
+          hadolint generic/Dockerfile.cuda.melodic
+          hadolint crossbuild/Dockerfile.kinetic-crossbuild
+          hadolint crossbuild/Dockerfile.kinetic-crossbuild-driveworks
+          hadolint crossbuild/Dockerfile.melodic-crossbuild
+          hadolint crossbuild/Dockerfile.melodic-crossbuild-driveworks
+
+  build-base:
+
+    if: github.repository_owner == 'sgermanserrano'
+
+    needs: dockerfile-lint
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, self-hosted]
+      fail-fast: false
+    env:
+      IMAGE_NAME: autoware/autoware
+      ROS_DISTRO: melodic
+      TAG_PREFIX: bleedingedge
+      VERSION: master
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Image build
+        run: |
+          docker login -u ${{ secrets.DKR_USR }} -p ${{ secrets.DKR_PASS }}
+          cd generic
+          ./build.sh -b -c off -i $IMAGE_NAME -r $ROS_DISTRO -t $TAG_PREFIX -v $VERSION
+
+      - name: amd64 image re-tag and push
+        if: contains(matrix.os, 'ubuntu-latest')
```

Comment:
Does the `matrix.os` variable get filtered based on where the job is running? It seems like this would be true no matter what.

Reply:
`matrix.os` is either `ubuntu-latest` which runs on Github infrastructure or `self-hosted` as defined above, hence only true for x86 runner on Actions runner